### PR TITLE
Only store `_edit_use_anchors_` metadata if value is not the default

### LINKED ISSUE #57862: Unnecessary _edit_use_anchors_
### Godot version

v4.0.alpha2

### System information

Windows 10

### Issue description

In both 3 and 4 creating a new control scene and saving produces something like this:

	[gd_scene format=3 uid="uid://bv8p1x3edr6dl"]

	[node name="Control" type="Control"]
	anchor_right = 1.0
	anchor_bottom = 1.0
	__meta__ = {
	"_edit_use_anchors_": false
	}

1.  Why is edit_use_anchors set to false when achors are clearly being used?
2.  Why is this set without interacting with the anchor button?
3.  Is there some way to opt out of this "feature". For those of us who do not want to attempt to store that menu setting in each node? I have not used this button once since installing, but renoved hundreds of "_edit_use_anchors_" triptics from my scene files to remove clutter and improve readability

### Steps to reproduce

1. Create new Control scene and save.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Calinou**
`master` version of https://github.com/godotengine/godot/pull/58254.

The default value is assumed to be `false`, so this metadata only needs to be stored if the value is `true`.

This closes https://github.com/godotengine/godot/issues/57862.

*Bugsquad edit (keywords for easier searching): edit_use_anchors*

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -4129,9 +4129,13 @@ void CanvasItemEditor::_set_anchors_and_offsets_to_keep_ratio() {
 			undo_redo->add_do_method(control, "set_anchor", SIDE_BOTTOM, bottom_right_anchor.y, false, true);
 			undo_redo->add_do_method(control, "set_meta", "_edit_use_anchors_", true);
```
Comment (akien-mga) [inline editor/plugins/canvas_item_editor_plugin.cpp:4130]: What about this?

Comment (Calinou) [reply editor/plugins/canvas_item_editor_plugin.cpp:4130]: It always stores the value `true`, so I believe it should be left in.

**Comment by rakkarage** [comment]
The pull request build seems to work for me. New scenes no longer have that setting written into them it seems.

But during testing I ran into another issue. You cannot undo a toggle of the anchor button? Maybe I will open a new issue?

Thanks.

**Comment by Calinou** [comment]
> But during testing I ran into another issue. You cannot undo a toggle of the anchor button? Maybe I will open a new issue?

Does this happen without this PR?

**Comment by rakkarage** [comment]
Yes, in 3 and 4.

- Make new Control scene and move control then toggle use anchors
- Scene->undo un-moves the control instead of the toggle

**Comment by Calinou** [comment]
> Yes, in 3 and 4.
> 
>     * Make new Control scene and drag/move control then toggle use anchors
> 
>     * Scene->undo un-moves the control instead of the toggle

Use Anchors is an editor-only action, so I don't think it makes sense to be tracked in undo/redo.

**Comment by rakkarage** [comment]
ya sorry I quess i still thought it was doing the the `_edit_use_anchors_` stuff :)

**Comment by aaronfranke** [comment]
Really this should be moved out of the scene file. We don't need to store local editor specifics in version-controlled scene files, this kind of data would have a better home in the `.godot/editor/` folder.

**Comment by pycbouh** [comment]
> Really this should be moved out of the scene file. We don't need to store local editor specifics in version-controlled scene files, this kind of data would have a better home in the `.godot/editor/` folder.

Do we have examples of other per-node properties stored in there? This is similar to `_edit_lock_`, which is also stored with nodes at the moment. There is also a question of how and if we should allow editing this programmatically, so that, say, a plugin could create a node and mark it to use anchors by default.

I agree in principle, but there is also an argument to be made that this needs to be synced and needs to be attached to the node.

**Comment by pycbouh** [comment]
This PR needs a rebase/remake since the relevant code is now in `control_editor_plugin.cpp`. It should mostly be the same, IIRC, but I'd still appreciate an update before reviewing it.

**Review by YuriSizov — APPROVED** [review]
The change seems to be straightforward.

@rakkarage has a point though. In one case we save this editor-only flag as a part of UndoRedo, and in another we don't. But that's existing behavior, so I think it's a non-blocker for this PR.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Could use a dedicated backport for `3.x` as this code has changed somewhat.

**Pull request merged**